### DISCUSSION TITLE
Integrating Elastic Backend

### DISCUSSION THREAD

**Discussion by WildDogOne** [certeu/droid#5] [Ideas]
I have been looking into the code a bit, and it seems like it would be easy to either integrate EQL (Event Query Language) or ES|QL.
But I have some logistical questions before I start going too deep into this.

EQL has a pysigma parse which can convert the Sigma rules to a fully described Elastic Security rule, with all parameters needed.
However EQL does not support correlations.

ES|QL does support correlations but not security rules.

So my idea would be to actually only use ESQL and then write an exporter that can handle all attributed needed by the elastic security component to run the query.

Does this make sense?
Or would it make more sense to have both backends?
Thoughts and ideas welcome :)

**Comment by 0xFustang** [comment]
Hello @WildDogOne! I believe we should go down this path. 

Having two exporters for elastic could make sense: `elastic-esql` and `elastic-eql`. Actually this matches the backend selection in the [pySigma-backend-elasticsearch](https://github.com/SigmaHQ/pySigma-backend-elasticsearch).

**Reply by WildDogOne** [reply]
OK that helped me already, I will build up both, but there should be a disclaimer that esql is to be prefered always, since it supports more sigma features compared to eql
